src/poetry/console/commands/check.py

```diff
@@ -95,6 +95,7 @@ def _validate_readme(self, readme: str | list[str], poetry_file: Path) -> list[s
     def _validate_dependencies_source(self, config: dict[str, Any]) -> list[str]:
         """Check dependencies's source are valid"""
         sources = {k["name"] for k in config.get("source", [])}
+        sources.add("pypi")
```

Comment:
PyPI might be deactivated. I think we should use `self.poetry.pool.all_repositories` to get the names of the sources.